src/base1/cockpit.js

```diff
@@ -3561,8 +3563,16 @@ function factory() {
             function try_read() {
                 read_channel = cockpit.channel(opts);
                 var content_parts = [ ];
+                var content_size = 0;
                 read_channel.addEventListener("message", function (event, message) {
-                    content_parts.push(message);
+                    content_size += message.length;
+                    if (content_size > max_read_size) {
+                        if (options.allow_partial_read)
+                            read_channel.close();
+                        else
+                            read_channel.close("file too large");
```

Comment:
I think this should have an actual problem code "too-large" ... and a general message should be added to ```cockpit.message()```.